### PR TITLE
Improved the coverage of ParseExact with the DateTimeStyle 

### PR DESCRIPTION
**Pull request by tidusjar**
Better coverage with AllowWhiteSpaces now as requested as part of #27679

### CODE REVIEW

File: src/System.Runtime/tests/System/DateTimeTests.cs [hunk 1]
```diff
@@ -1352,6 +1352,68 @@ public static IEnumerable<object[]> ParseExact_ValidInput_Succeeds_MemberData()
...
+            yield return new object[] { "1234-05-06T07:00:00Z", "yyyy-MM-dd'T'HH:mm:ss.FFF'Z'\"  \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, new DateTime(1234, 5, 6, 7, 0, 0, 0) };
+            yield return new object[] { "1234-05-06T07:00:00Z", "yyyy-MM-dd'T'HH:mm:ssFFF'Z'\' \'", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, new DateTime(1234, 5, 6, 7, 0, 0, 0) };
+            yield return new object[] { "1234-05-06T07:00:00Z", "yyyy-MM-dd'T'HH:mm:ssFFF'Z'\'  \'", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, new DateTime(1234, 5, 6, 7, 0, 0, 0) };
+            yield return new object[] { "1234-05-06T07:00:00Z", "yyyy-MM-dd'T'HH:mm:ssFFFZ\" \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, TimeZoneInfo.ConvertTimeFromUtc(new DateTime(1234, 5, 6, 7, 0, 0, DateTimeKind.Utc), TimeZoneInfo.Local) };
+            yield return new object[] { "1234-05-06T07:00:00GMT", "yyyy-MM-dd'T'HH:mm:ssFFFZ\"  \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, TimeZoneInfo.ConvertTimeFromUtc(new DateTime(1234, 5, 6, 7, 0, 0, DateTimeKind.Utc), TimeZoneInfo.Local) };
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/DateTimeTests.cs]: Thanks for adding these, but most of these don't actually have any whitespace in them. Is that intentional?

Comment (stephentoub) [reply src/System.Runtime/tests/System/DateTimeTests.cs]: To actually test AllowWhiteSpace, we need to verify correct behavior both with and without whitespace, and when with it, in various kinds, amounts, and locations. AllowWhiteSpace allows many formats to have arbitrary whitespace at arbitrary locations, and I believe the intent of the associated issue is to validate that, but @ahsonkhan can correct me if that's wrong.

Comment (ahsonkhan) [reply src/System.Runtime/tests/System/DateTimeTests.cs]: > AllowWhiteSpace allows many formats to have arbitrary whitespace at arbitrary locations, and I believe the intent of the associated issue is to validate that

That's correct.

Comment (tidusjar) [reply src/System.Runtime/tests/System/DateTimeTests.cs]: Ok I'll make the appropriate changes! Thanks for the clarification!

**Comment by stephentoub** [comment]
@tidusjar, should we close this PR and you'll open a new one if/when you decide to revamp your approach?  Or do you want to just update this PR?

**Comment by tidusjar** [comment]
Sorry for the delay, i'll update this PR soon.

**Comment by stephentoub** [comment]
@dotnet-bot test Linux x64 Release Build please
@dotnet-bot OSX x64 Debug Build please

Infrastructure problems.

**Comment by stephentoub** [comment]
@dotnet-bot test OSX x64 Debug Build please

**Comment by tidusjar** [comment]
Yeah I did notice they timed out. Was unsure how to restart. Sorry!

**Comment by stephentoub** [comment]
> Sorry!

No need to apologize for anything.  We're sorry sometimes our CI infrastructure is flaky like that.

**Comment by stephentoub** [comment]
Thanks for updating.  Are the tests in the first commit still needed?  Do you they add coverage for things not already covered?  Seems like the second commit is the actual value now.

**Comment by tidusjar** [comment]
I think the tests in the First commit verify there are no different behaviour when there is no whitespace. 

But if you feel like they are redundant I can remove.

**Comment by tidusjar** [comment]
That is a better solution, ensures better coverage and less code duplication.

Made those changes.

File: src/System.Runtime/tests/System/DateTimeTests.cs [hunk 2]
```diff
@@ -1352,6 +1354,65 @@ public static IEnumerable<object[]> ParseExact_ValidInput_Succeeds_MemberData()
...
+            yield return new object[] { "2001", "yyyy\" \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, new DateTime(2001, 1, 1, 0, 0, 0) };
+
+            yield return new object[] { "3", "%H\"  \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, DateTime.Today + TimeSpan.FromHours(3) };
+            yield return new object[] { "03", "HH\" \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, DateTime.Today + TimeSpan.FromHours(3) };
+
+            yield return new object[] { "3A", "ht\"  \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, DateTime.Today + TimeSpan.FromHours(3) };
+            yield return new object[] { "03A", "hht\" \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, DateTime.Today + TimeSpan.FromHours(3) };
+            yield return new object[] { "3P", "ht\'  \'", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, DateTime.Today + TimeSpan.FromHours(12 + 3) };
+            yield return new object[] { "03P", "hht\" \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, DateTime.Today + TimeSpan.FromHours(12 + 3) };
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/DateTimeTests.cs]: Why do you still need all of these?  Aren't they now duplicates?

**Comment by tidusjar** [comment]
Sorry I had no idea, I thought I removed them, obviously not.

**Comment by tidusjar** [comment]
@dotnet-bot test Tizen armel Debug Build please

File: src/System.Runtime/tests/System/DateTimeTests.cs [hunk 3]
```diff
@@ -1352,6 +1354,44 @@ public static IEnumerable<object[]> ParseExact_ValidInput_Succeeds_MemberData()
             yield return new object[] { "1234-05-06T07:00:00GMT", "yyyy-MM-dd'T'HH:mm:ssFFFZ\"  \"", CultureInfo.InvariantCulture, DateTimeStyles.AllowTrailingWhite, TimeZoneInfo.ConvertTimeFromUtc(new DateTime(1234, 5, 6, 7, 0, 0, DateTimeKind.Utc), TimeZoneInfo.Local) };
 
 
+            yield return new object[] { "9", "\"  \"%d", CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, new DateTime(DateTime.Now.Year, 1, 9, 0, 0, 0) };
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/DateTimeTests.cs]: I'm confused... these all still appear to be here?

**Comment by tidusjar** [comment]
Sorry I don't know what's gotten into me! Cleaned them up now.

**Pull request merged**